operator/pkg/controllers/master/kubeconfigs.go

```diff
@@ -46,98 +62,108 @@ func (c *Controller) reconcileKubeConfigs(ctx context.Context, controlPlane *v1a
 			clusterName: {
-				Server:                   endpoint,
+				Server:                   fmt.Sprintf("https://%s:443", request.endpoint),
 				CertificateAuthorityData: caCert,
 			},
 		},
 		Contexts: map[string]*clientcmdapi.Context{
 			contextName: {
 				Cluster:  clusterName,
-				AuthInfo: userName,
+				AuthInfo: request.auth.Name,
 			},
 		},
 		AuthInfos: map[string]*clientcmdapi.AuthInfo{
-			userName: {
+			request.auth.Name: {
 				ClientKeyData:         key,
 				ClientCertificateData: cert,
 			}},
 		CurrentContext: contextName,
 	}
 }
 
-func kubeAdminCertConfig(clusterName string) *secrets.Request {
-	return &secrets.Request{
-		Name: kubeAdminSecretNameFor(clusterName),
-		Config: &certutil.Config{
-			Usages:       []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
-			CommonName:   "kubernetes-admin",
-			Organization: []string{"system:masters"},
-		},
-	}
+type configRequest struct {
+	endpoint string
+	auth     *secrets.Request
 }
 
-func kubeSchedulerCertConfig(clusterName string) *secrets.Request {
-	return &secrets.Request{
-		Name: kubeSchedulerSecretNameFor(clusterName),
-		Config: &certutil.Config{
-			Usages:     []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
-			CommonName: "system:kube-scheduler",
-		},
-	}
+func kubeAdminCertConfig(clusterName, endpoint string, caSecret *v1.Secret) *configRequest {
+	return &configRequest{
+		endpoint: endpoint,
+		auth: &secrets.Request{
+			Name:     kubeAdminSecretNameFor(clusterName),
+			Type:     secrets.KeyWithSignedCert,
+			CASecret: caSecret,
+			Config: &certutil.Config{
+				Usages:       []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+				CommonName:   "kubernetes-admin",
+				Organization: []string{"system:masters"},
+			},
+		}}
 }
 
-func kubeControllerManagerCertConfig(clusterName string) *secrets.Request {
-	return &secrets.Request{
-		Name: kubeControllerManagerSecretNameFor(clusterName),
-		Config: &certutil.Config{
-			Usages:     []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
-			CommonName: "system:kube-controller-manager",
-		},
-	}
+func kubeSchedulerCertConfig(clusterName, endpoint string, caSecret *v1.Secret) *configRequest {
+	return &configRequest{
+		endpoint: endpoint,
+		auth: &secrets.Request{
+			Name:     kubeSchedulerSecretNameFor(clusterName),
+			Type:     secrets.KeyWithSignedCert,
+			CASecret: caSecret,
+			Config: &certutil.Config{
+				Usages:     []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+				CommonName: "system:kube-scheduler",
+			},
+		}}
+}
+
+func kubeControllerManagerCertConfig(clusterName, endpoint string, caSecret *v1.Secret) *configRequest {
+	return &configRequest{
+		endpoint: endpoint,
+		auth: &secrets.Request{
+			Name:     kubeControllerManagerSecretNameFor(clusterName),
+			Type:     secrets.KeyWithSignedCert,
+			CASecret: caSecret,
+			Config: &certutil.Config{
+				Usages:     []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+				CommonName: "system:kube-controller-manager",
+			},
+		}}
```

Comment:
bit awkward to have the last line of this be }}. If you're going with this style (I'm not really a fan), I'd include the } on line 165, too.